tests/unit/test-wcs-notifications-debug-tool.php

```diff
@@ -212,4 +155,78 @@ public function test_tool_state_while_processing() {
 		$this->assertIsArray( $tool_state );
 		$this->assertFalse( isset( $tool_state['last_offset'] ) );
 	}
+
+	/**
+	 * Data provider for the "test_process_batch_notifications()" method.
+	 *
+	 * @return array
+	 */
+	protected function notification_subscription_data_provider() {
+
+		/*
+		 * Create a simple subscription.
+		 */
+		$simple_subscription = WCS_Helper_Subscription::create_subscription(
+			[
+				'billing_period'   => 'month',
+				'billing_interval' => 1,
+			]
+		);
+
+		$simple_subscription->update_status( 'active' );
+		$simple_subscription->save();
+
+		$simple_subscription->update_dates(
+			[
+				'next_payment' => '2034-09-20 08:08:08',
```

Comment:
Is this supposed to be "always" in the future? Why not use a relative date, then?

Reply:
Also, is it a problem that 1-month subscription has next payment in 10 year's time? :)

Reply:
> Is this supposed to be "always" in the future? Why not use a relative date, then?

Yes, I suppose I can make these dates dynamic. I wouldn't want someone 10 years from now cursing me while digging into a nasty unit test bug. 😅 

> Also, is it a problem that 1-month subscription has next payment in 10 year's time? :)

Yes, it seems odd. I believe it's related to our recent conversation about the differences between a "checkout" flow and the freestyle data model, which I actually don't mind.

Reply:
Done https://github.com/Automattic/woocommerce-subscriptions-core/pull/689/commits/f9bee572a8cdd2f32271bfa2a4a05b743ca7ba54 and https://github.com/Automattic/woocommerce-subscriptions-core/pull/689/commits/725f13334e230eec6cfc0352dd123c5b111ba232